docs: Update Contribution guide

princerachit opened the pull request:
## Description
<!-- Describe your changes in detail -->
Update contribution guide and readme.

## Related Issue(s)
<!-- List the issue(s) this PR solves -->
Fixes https://github.com/gitpod-io/workspace-images/issues/572

## How to test
<!-- Provide steps to test this PR -->
You can navigate to the branch and open the README.md and the CONTRIBUTING.md file to check if the text renders correctly.

## Release Notes
<!--
  Add entries for the CHANGELOG.md or "NONE" if there aren't any user facing changes.
  Each line becomes a separate entry.
  Format: [!<optional for breaking>] <description>
  Example: !basic auth is no longer supported
  See https://www.notion.so/gitpod/Release-Notes-513a74fdd23b4cb1b3b3aefb1d34a3e0
-->
```release-note
Update contribution guide
```

## Documentation
<!--
Does this PR require updates to the documentation at www.gitpod.io/docs?
* Yes
  * 1. Please create a docs issue: https://github.com/gitpod-io/website/issues/new?labels=documentation&template=DOCS-NEW-FEATURE.yml&title=%5BDocs+-+New+Feature%5D%3A+%3Cyour+feature+name+here%3E
  * 2. Paste the link to the docs issue below this comment
* No
  * Are you sure? If so, nothing to do here.
-->

princerachit commented on CONTRIBUTING.md:
```diff
@@ -1,58 +1,159 @@
+1. Write your tests with proper assertions.
+You can read more on how to write tests in the [dazzle documentation](https://github.com/gitpod-io/dazzle/#testing-layers-and-merged-images)
+
+Here is a list of best practices you should keep in mind while adding a chunk:
```
```suggestion
Here is a list of best practices you should keep in mind while writing a test:
```

Furisto commented on CONTRIBUTING.md:
```diff
@@ -1,58 +1,171 @@
+e.g. `ENV PATH=/my-tool/path/bin:$PATH`.
+Not doing so can cause path conflicts and can potentially break other images.
+1. **DO NOT** update the default shell rc files like `.bashrc` unless you are making change in the base layer.
+1. Always create an rc file in the `.bashrc.d` directory.
```
At first glance this sounded like you always need to create an rc file.
```suggestion
1. If you need to create an rc file you should put it in the `.bashrc.d` directory.
```

Furisto commented on CONTRIBUTING.md:
```diff
@@ -1,58 +1,171 @@
+e.g. `ruby-3`
+1. Use `base` combination as reference rather than other combination unless there is an exception.
+e.g. `full-vnc` is one of the cases which uses the `full` combination rather than the `base`.
+This is because combination as the vnc combination only adds an extra `tool-vnc` chunk and nothing else.
```
```suggestion
This is because the vnc combination only adds an extra `tool-vnc` chunk and nothing else.
```

princerachit commented on CONTRIBUTING.md:
````diff
@@ -1,58 +1,171 @@
 ```
 
-That's it. Simply pushing the Pull Request's commits to a new branch under `upstream` is enough to trigger CircleCI for that PR. Pretty soon the build status will appear in the Pull Request, automatically.
+where `combo` is the name of the combination defined in [dazzle.yml](dazzle.yml) e.g. `full`.
````
Nice catch. There is a typo.

princerachit commented:
👋🏾 @Furisto , thanks for reviewing this. I have addressed your comments. After reading this guide would you as a first time contributor have confidence to contribute?

Furisto commented:
@princerachit I would. I cannot think of anything else that I would like to see in the guide at the moment.
